tests/src/JIT/HardwareIntrinsics/X86/Shared/GenerateTests.csx

```diff
@@ -1243,7 +1341,22 @@ private static void ProcessInput(StreamWriter testListFile, string groupName, (s
     }
 
     var testFileName = Path.Combine("..", groupName, $"{testName}.cs");
-    var template = File.ReadAllText(input.templateFileName);
+    var matchingTemplate = Templates.Where((t) => t.outputTemplateName.Equals(input.templateFileName)).SingleOrDefault();
```

Comment:
This basically does two passes now. The first pass gets the shared template (if one exists and falling back to the explicit template otherwise) and adds in the specialized template logic. The second pass works exactly as it did before and adds in the test specific logic.